### PR TITLE
Extract error message from libhdfs3

### PR DESCRIPTION
**Pull request by jettify**
Error messages right now is not informative especially  connection errors. I have added error extracting logic from libhdfs3

### CODE REVIEW

**Comment by mrocklin** [comment]
Very cool.  I actually was curious about how to get error messages out of libhdfs3.  

Can I ask you to add a quick test that verifies that we're actually getting an informative error message out from at least one of the functions?  I usually test this with a try-except block as follows:

```python
try:
    some_illegal_action()
except Exception as e:
    assert 'some special text' in str(e)
```

**Comment by jettify** [comment]
Sure, I will add tests but bit later today.

FYI `hdfs3` and `libhdfs3` works fine with kerberos if one pass `pars={'hadoop.security.authentication': 'kerberos'}` option.

**Comment by jettify** [comment]
done

File: hdfs3/tests/test_hdfs3.py [hunk 1]
```diff
@@ -48,6 +45,14 @@ def test_simple(hdfs):
...
         assert out == data
 
+def test_connection_error():
+    with pytest.raises(RuntimeError) as ctx:
+        hdfs = HDFileSystem(host='localhost', port=9999, connect=False)
+        hdfs.connect()
+    # error message is long and with java exceptions, so here we just check
+    # that important part of error is present
+    msg = 'Caused by: HdfsNetworkConnectException: Connect to "localhost:9999"'
+    assert msg in str(ctx.value)
```
Comment (mrocklin) [inline hdfs3/tests/test_hdfs3.py]: Oh neat, I had no idea `pytest.raises` offered this functionality.

**Comment by mrocklin** [comment]
+1 

Merging soon if no comments.

**Pull request merged**